Electrokinetics tutorial

reinaual opened the pull request:
Write diffusion-advection-reaction tutorial.

christophlohrmann commented on doc/tutorials/electrokinetics/electrokinetics.ipynb:
```diff
@@ -4,385 +4,708 @@
+    "    ext_efield=[0., 0., 0.], tau=TAU)\n",
+    "system.ekcontainer.add(species)\n",
+    "\n",
+    "species[BOX_L[0] // 2, BOX_L[1] // 2, 0].density = 1.0"
```
I would put 1/agrid**3 here to make sure there is one unit of material in the system.
Alternatively would using a gaussian as the starting condition reduce numerical errors connected to discretised gradients?

christophlohrmann commented on doc/tutorials/electrokinetics/electrokinetics.ipynb:
```diff
@@ -4,385 +4,708 @@
+   },
+   "outputs": [],
+   "source": [
+    "total_frames = 200\n",
```
the video takes way too long to be produced. While the simulation is running, students can't use their interactive session to do subsequent tasks. Could the order be switched so this is last? It also seems like the most advanced system.

christophlohrmann commented on doc/tutorials/electrokinetics/electrokinetics.ipynb:
```diff
@@ -4,385 +4,708 @@
+    "    progress_bar.update()\n",
+    "    \n",
+    "\n",
+    "animation.FuncAnimation(fig, draw_frame, frames=range(total_frames), interval=300)"
```
would be cool if the boundaries were visible in the video

christophlohrmann reviewed (changes requested):
The physics in the tutorial semms good, especially the reaction in turbulent flow looks fantastic. Obviously, the exercises are still missing.
The language/style of the descriptive text needs to be improved, but that's something I could do

christophlohrmann reviewed (commented):
Exercises are a bit short (especially in the summer school context where we need people to do stuff for 90mins) but cover the essentials.
Text still needs some polishing. I could take it from here

christophlohrmann commented:
@jngrad @reinaual ready for review

christophlohrmann reviewed (approved):
can I approve my own changes?

reinaual committed:
New EK tutorial

christophlohrmann committed:
polish text and exercises

jngrad committed:
Rework electrokinetics tutorial

Write tests to check for analytical solutions and turbulent flow.
Improve matplotlib figures, ffmpeg videos and png figure captions.
Fix undefined behavior during integration (missing LB thermostat).
